Multi output

NiklasHansen opened the pull request:
Prepared for multiple output types (Issue #4), and added CSV support.
I created an IOutput interface, and abstracted both Slack and Console into IOutput. Then I created another IOutput to support CSV files.

NiklasHansen committed:
Created IOutput interface, for generic output forms (Slack, Console included)

NiklasHansen committed:
Modified test suite to suit new setup using IOutput

NiklasHansen committed:
Renamed method, as it no longer affects Slack directly

NiklasHansen committed:
Added CSV output

hmol commented:
Great work! This is awsome :+1: 
3 things I'm wondering:
1. Do you think it could be useful to have the delimiter of the csv file configurable? I know comma is the most used, but I also know some people like to use semicolon. Something like `<add key="Csv.Delimiter" value="," />` in the App.config
2. The code in the constructor of CsvOutput.cs, could it be moved into a private method? Like this `private void SetUp()`. I'm a bit quirky when it comes to code in constructors, sorry :neckbeard:
3. The readme file needs to be updated with info about the new output for csv. Especially the table that explains App.settings must contain the new settings for csv.

Thanks for contributing to my little repo here :smiley:

NiklasHansen committed:
Incorporated changes from @hmol

NiklasHansen committed:
Updated README

NiklasHansen commented:
I completely agree with making the delimiter a configuration. I meant to do that, but did not get around to do it. I have created another commit with this change.

I have moved the setup to a private method - I agree with it not being "pretty" with all that code in the constructor.

I have also updated the readme, to contain the new app-settings.

hmol commented:
Thanks a bunch :grinning:
